packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp

```diff
@@ -78,7013 +73,9090 @@
         (Wachieved) / (wExpected) - 1
 
-
 #define ZOLTAN2_ALGMULTIJAGGED_SWAP(a,b,temp) temp=(a);(a)=(b);(b)=temp;
 
-
 namespace Teuchos{
 
 /*! \brief Zoltan2_BoxBoundaries is a reduction operation
- * to all reduce the all box boundaries.
+ *  to all reduce the all box boundaries.
 */
-
 template <typename Ordinal, typename T>
 class Zoltan2_BoxBoundaries  : public ValueTypeReductionOp<Ordinal,T>
 {
 private:
-    Ordinal size;
-    T _EPSILON;
+  Ordinal size;
+  T _EPSILON;
 
 public:
-    /*! \brief Default Constructor
-     */
-    Zoltan2_BoxBoundaries ():size(0), _EPSILON (std::numeric_limits<T>::epsilon()){}
-
-    /*! \brief Constructor
-     *   \param nsum  the count of how many sums will be computed at the
-     *             start of the list.
-     *   \param nmin  following the sums, this many minimums will be computed.
-     *   \param nmax  following the minimums, this many maximums will be computed.
-     */
-    Zoltan2_BoxBoundaries (Ordinal s_):
-        size(s_), _EPSILON (std::numeric_limits<T>::epsilon()){}
-
-    /*! \brief Implement Teuchos::ValueTypeReductionOp interface
-     */
-    void reduce( const Ordinal count, const T inBuffer[], T inoutBuffer[]) const
-    {
-        for (Ordinal i=0; i < count; i++){
-            if (Z2_ABS(inBuffer[i]) >  _EPSILON){
-                inoutBuffer[i] = inBuffer[i];
-            }
-        }
+  /*! \brief Default Constructor
+   */
+  Zoltan2_BoxBoundaries() : size(0),
+    _EPSILON(std::numeric_limits<T>::epsilon()) {}
+
+  /*! \brief Constructor
+   *  \param Ordinal          TODO: Documentaion
+   */
+  Zoltan2_BoxBoundaries(Ordinal s_):
+    size(s_), _EPSILON(std::numeric_limits<T>::epsilon()) {}
+
+  /*! \brief Implement Teuchos::ValueTypeReductionOp interface
+   *  \param count            TODO: Documentation
+   *  \param inBuffer         TODO: Documentation
+   *  \param inoutBuffer      TODO: Documentation
+   */
+  void reduce( const Ordinal count, const T inBuffer[], T inoutBuffer[]) const {
+    for(Ordinal i = 0; i < count; i++) {
+      if(Z2_ABS(inBuffer[i]) > _EPSILON) {
+        inoutBuffer[i] = inBuffer[i];
+      }
     }
+  }
 };
+
 } // namespace Teuchos
 
 namespace Zoltan2{
 
 /*! \brief Allocates memory for the given size.
- *
+ *  \param size   size of allocation
  */
 template <typename T>
-T *allocMemory(size_t size){
-    if (size > 0){
-        T * a = new T[size];
-        if (a == NULL) {
-            throw  "cannot allocate memory";
-        }
-        return a;
-    }
-    else {
-        return NULL;
+T *allocMemory(size_t size) {
+  if(size > 0) {
+    T * a = new T[size];
+    if(a == NULL) {
+      throw  "cannot allocate memory";
     }
+    return a;
+  }
+  else {
+    return NULL;
+  }
 }
 
 /*! \brief Frees the given array.
- *
  */
 template <typename T>
-void freeArray(T *&array){
-    if(array != NULL){
-        delete [] array;
-        array = NULL;
-    }
+void freeArray(T *&array) {
+  if(array != NULL) {
```

Comment:
You don't need this.  `delete` on a null pointer is harmless.  https://en.cppreference.com/w/cpp/language/delete

Reply:
Fixed. @kddevin I think just using new and delete[] would be a good change but I wasn't positive if there might be a logic problem if something allocated for size 0. Original would return null but now we have new type[0] returns a non null ptr in C++. I kept this change as a separate commit in case it was a problem. But tests are all ok.